# Blast Theobroma cacao Datasets vs. Drought DB
<br>Nucleotide-Nucleotide BLAST 2.5.0+
- Blast+ default settings 
<br>Gap Penalties: Existence: 0, Extension: 2.5
<br>evalue:  Default = 10'

## A. Expressed Sequence Tag (EST): Transcriptome
Blasting Theobroma cacao EST (**163,304 sequences**) from Argout et al. 2008 (EMBL DataBase CU469588 to CU633156), from several plant tissue, against a set of 194 Drought related genes from DroughtDB.

__Aim:__ Estimate and explore how many of the drought genes will show up in the Guazuma transcriptome.

### Test 1.1: Theobroma cacao EST (163,304 sequences)

__1. Create a custom database from the multi-FASTA EST file of sequences with this minimal command:__

In [15]:
makeblastdb -in CU469588-CU633156_all.fasta -dbtype nucl



Building a new DB, current time: 07/01/2020 14:48:17
New DB name:   /media/storage/OneDrive - University of Edinburgh/onedrive_storage/Guazuma/Genomics/Bait_design/notebook_jup/CU469588-CU633156_all.fasta
New DB title:  CU469588-CU633156_all.fasta
Sequence type: Nucleotide
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 163304 sequences in 13.2114 seconds.


__2. Perform the blast:__

Query: 194 sequences
Database: 163,304 sequences

In [17]:
blastn -db CU469588-CU633156_all.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results.out

__3. Use BLASTGrabber__, a Java application, locally installed and with a user-friendly GUI to visualize BLAST output data (https://www.mn.uio.no/ibv/english/people/aca/kamran/software/blastgrabber/).

__RESULTS summary:__
- Total number of hits: 683
- Total number of unique hits: 628
- Max number of hits for a query (AY857933.1): 250
- Min number of hits for a query: 1
- 25/194 query sequences had at least 1 hit


### Test 1.2: Filtered ESTs for leaves, shoots, root-drought 


17 libraries (filtered files): **36,925**

- DROUGHTLS_KZ0ACAF: 2766 seqs
- LEAVES_KZ0ABE: 4695 seqs
- DROUGHTRS_KZ0ACAE: 2685 sequences (Roots under drought stress)
- COSSHPPI_KZ0AA and COSSHPPS_KZ0AA: COSSHPPI_IS_KZ0AA_fixed 10985 sequences
- LEAVPAGU_KZ0ACQ: 3030 sequences
- LEPAPNGR_KZ0ACP: 1021 sequences
- LESSHMEPNGa_KZ0ACAP: 356 sequences
- LESSHMEPNGb_KZ0ACV: 1244 sequences
- LESSHPNGRSb_KZ0ABP: 701 sequences
- MIRIDUFS_KZ0ACAD: 3007 sequences
- RESSHMEPNGb_KZ0AC_fixed: 1285 sequences
- RPPSSHPNGa_KZ0ACAL: 344 sequences
- RPPSSHPNGb_KZ0ACR: 1406 sequences
- SSHMIRUFa_KZ0ACAN: 318 sequences
- SSHMIRUFb_KZ0ACT: 1391 sequences
- STSSHWB1S_KZ0ABI_K: 283 sequences
- STSSHWB2Sb_KZ0ACB: 1408 sequences

__1. Create database__

In [21]:
makeblastdb -in STSSHWB2Sb_KZ0ACB.fasta -dbtype nucl



Building a new DB, current time: 07/09/2020 11:54:51
New DB name:   /media/storage/OneDrive - University of Edinburgh/onedrive_storage/Guazuma/Genomics/Bait_design/notebook_jup/STSSHWB2Sb_KZ0ACB.fasta
New DB title:  STSSHWB2Sb_KZ0ACB.fasta
Sequence type: Nucleotide
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 1408 sequences in 0.157897 seconds.


__2. Perform Blast__

Query: 194 sequences DroughtDB

In [3]:
blastn -db Drought_KZ0ACAF.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_DROUGHTLS_KZ0ACAF.out

blastn -db LEAVES_KZ0ABE.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_LEAVES_KZ0ABE.out

In [23]:
blastn -db DROUGHTRS_KZ0ACAE.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_DROUGHTRS_KZ0ACAE.out

blastn -db COSSHPPI_IS_KZ0AA_fixed.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_COSSHPPI_IS_KZ0AA_fixed.out

blastn -db LEAVPAGU_KZ0ACQ.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_LEAVPAGU_KZ0ACQ.out

blastn -db LEPAPNGR_KZ0ACP.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_LEPAPNGR_KZ0ACP.out

blastn -db LESSHMEPNGa_KZ0ACAP.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_LESSHMEPNGa_KZ0ACAP.out

blastn -db LESSHMEPNGb_KZ0ACV.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_LESSHMEPNGb_KZ0ACV.out


blastn -db LESSHPNGRSb_KZ0ABP.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_LESSHPNGRSb_KZ0ABP.out

blastn -db MIRIDUFS_KZ0ACAD.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_MIRIDUFS_KZ0ACAD.out

blastn -db RESSHMEPNGb_KZ0AC_fixed.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_RESSHMEPNGb_KZ0AC_fixed.out

blastn -db RPPSSHPNGa_KZ0ACAL.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_RPPSSHPNGa_KZ0ACAL.out

blastn -db RPPSSHPNGb_KZ0ACR.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_RPPSSHPNGb_KZ0ACR.out


blastn -db SSHMIRUFa_KZ0ACAN.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_SSHMIRUFa_KZ0ACAN.out

blastn -db SSHMIRUFb_KZ0ACT.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_SSHMIRUFb_KZ0ACT.out


blastn -db STSSHWB1S_KZ0ABI_K.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_STSSHWB1S_KZ0ABI_K.out


blastn -db STSSHWB2Sb_KZ0ACB.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_STSSHWB2Sb_KZ0ACB.out

__3. Use BlastGrabber__


__RESULTS__ DROUGHTLS_KZ0ACAF Summary:
- Total number of hits: 78
- Total number of unique hits: 78
- Highest number for a given query (AY857933.1): 78
- Min number of hits for a query:0
- 1/194 query sequences had at least 1 hit (AY857933.1)


__RESULTS__ LEAVES_KZ0ABE Summary:
- Total number of hits: 6
- Total number of unique hits: 6
- Highest number for a given query: 3
- Min number of hits for a query: 1
- 4/194 query sequences had at least 1 hit



## B. Genome: Criollo Version 2
Blasting T. cacao Belizian Criollo B97-61/B2 Genome Version 2 from Argout et al. 2017 (Cocoa Genome Hub).


Aim: Run Blast with the Cacao Genome vs. Drought DB, to compare with the results obtained in the EST Blast.

### Test 2.1: CDS intron less
__1. Create database__

In [4]:
makeblastdb -in Theobroma_cacaoV2_annot_cds.fasta -dbtype nucl



Building a new DB, current time: 07/03/2020 13:03:13
New DB name:   /media/storage/OneDrive - University of Edinburgh/onedrive_storage/Guazuma/Genomics/Bait_design/notebook_jup/Theobroma_cacaoV2_annot_cds.fasta
New DB title:  Theobroma_cacaoV2_annot_cds.fasta
Sequence type: Nucleotide
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 30655 sequences in 2.26917 seconds.


__2. Perform the blast:__


Query: 194 sequences DroughtDB

Database: 30,655 sequences CDS


In [7]:
blastn -db Theobroma_cacaoV2_annot_cds.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_CDS.out

__3. Use BLASTGrabber__

__RESULTS__ Summary:
- Total number of hits: 59
- Total number of unique hits: 53
- Highest number for a given query: 8
- Min number of hits for a query (AT1G78580.1): 1
- 34/194 query sequences had at least 1 hit

### Test 2.2: Assembly scaffolds

__1. Create database__

In [11]:
makeblastdb -in Theobroma_cacao_criollo_scaffold.v2.0.fasta -dbtype nucl



Building a new DB, current time: 07/03/2020 14:58:12
New DB name:   /media/storage/OneDrive - University of Edinburgh/onedrive_storage/Guazuma/Genomics/Bait_design/notebook_jup/Theobroma_cacao_criollo_scaffold.v2.0.fasta
New DB title:  Theobroma_cacao_criollo_scaffold.v2.0.fasta
Sequence type: Nucleotide
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 554 sequences in 4.78315 seconds.


__2. Perform Blast__

Query sequences: 194 DroughtDB


Database: 554 sequences, scaffolds


In [13]:
blastn -db Theobroma_cacao_criollo_scaffold.v2.0.fasta -query completeCDS_DroughtDB_194Seqs.fasta -out results_Blast_Scaffolds.out

__3. Use BLASTGrabber__

__RESULTS__ Summary:
- Total number of hits: 43
- Total number of unique hits: 24
- Highest number for a given query (AT1G78580.1): 4
- Min number of hits for a query: 1
- 33/194 query sequences had at least 1 hit